stl/inc/ratio

```diff
@@ -19,59 +20,58 @@ _STL_DISABLE_CLANG_WARNINGS
 #undef new
 
 _STD_BEGIN
-template <intmax_t _Val>
-struct _Abs : integral_constant<intmax_t, (_Val < 0 ? -_Val : _Val)> {}; // computes absolute value of _Val
-
-template <intmax_t _Ax, intmax_t _Bx, bool _Sfinae = false,
-    bool _Good = (_Abs<_Ax>::value <= INTMAX_MAX / (_Bx == 0 ? 1 : _Abs<_Bx>::value))>
-struct _Safe_mult : integral_constant<intmax_t, _Ax * _Bx> {}; // computes _Ax * _Bx without overflow
+_NODISCARD constexpr intmax_t _Abs(const intmax_t _Val) noexcept { // computes absolute value sof _Val
+    return _Val < 0 ? -_Val : _Val; // caller ensures that _Val does not go below -INTMAX_MAX

```

Comment:
An internal assert is better than a comment imho
```suggestion
    _STL_INTERNAL_CHECK(_Val != INTMAX_MAX);
    return _Val < 0 ? -_Val : _Val;
```

Reply:
Then it should be `_STL_INTERNAL_CHECK(_Val != INTMAX_MIN);`

Reply:
`_STL_INTERNAL_CHECK` lives in `yvals.h`, and it cannot be included here because `<ratio>` is a core header.

Reply:
The function has been removed and calculations are inlined.